client/signup/difm/translation-hooks.tsx

```diff
@@ -50,3 +50,49 @@ export function useTranslatedPageTitles() {
 	);
 	return pages;
 }
+
+export function useTranslatedPageDescriptions() {
+	const translate = useTranslate();
+	const descriptions: Record< string, TranslateResult > = useMemo(
```

Comment:
Not sure we need to optimize this since `translate()` is already heavily optimised [using an LRU cache](https://github.com/Automattic/wp-calypso/blob/54ff4556c2e4eeae2810d0e93ae212cb4e3b0731/packages/i18n-calypso/src/i18n.js#L148). Having these values in a JS object is already a hash map and so we can simply return the plain object. Feel like this might be a bit of [unnecessary complexity](https://kentcdodds.com/blog/usememo-and-usecallback).

Reply:
TIl about the LRU cache! I followed the same pattern as `useTranslatedPageTitles` in the same file. But if the cache is in place, then we don't need useMemo in both hooks.